app/src/org/commcare/android/tasks/LogSubmissionTask.java

```diff
@@ -207,8 +207,15 @@ private boolean submit(DeviceReportRecord slr, int index) {
         //signal that it's time to start submitting the file
         this.startSubmission(index, f.length());
         HttpRequestGenerator generator;
-        User user = CommCareApplication._().getSession().getLoggedInUser();
-        if(user.getUserType().equals(User.TYPE_DEMO)) {
+        User user = null;
+        try {
+            user = CommCareApplication._().getSession().getLoggedInUser();
+        } catch (SessionUnavailableException e) {
+            // Session probably expired, so we will continue submitting logs,
+            // pretending to be a demo user.
+        }
+
+        if (user == null || user.getUserType().equals(User.TYPE_DEMO)) {

```

Comment:
Is it valid to submit logs as demo user if the session isn't found? Or should this just abort if the session has expired?